tests/data_files/Makefile

```diff
@@ -1369,6 +1369,18 @@ pkcs7_data_signed_badsigner.der: pkcs7_data_cert_signed_sha256.der
 	echo -en '\xa1' | dd of=$@ bs=1 seek=918 conv=notrunc
 all_final += pkcs7_data_signed_badsigner.der
 
+# pkcs7 signature file with invalid tag in signerInfo[1].serial after long issuer name
+pkcs7_signerInfo_1_serial_invalid_tag_after_long_name.der: pkcs7_data_multiple_signed.der
+	cp $< $@
+	echo -en '\xa1' | dd of=$@ bs=1 seek=498 conv=notrunc
+all_final += pkcs7_signerInfo_1_serial_invalid_tag_after_long_name.der
+
+# pkcs7 signature file with invalid tag in signerInfo[2]
+pkcs7_signerInfo_2_invalid_tag.der: pkcs7_data_3_signed.der
+	cp $< $@
+	echo -en '\xa1' | dd of=$@ bs=1 seek=810 conv=notrunc
```

Comment:
```suggestion
	echo 'a1' | xxd -p -r | dd of=$@ bs=1 seek=498 conv=notrunc
all_final += pkcs7_signerInfo_1_serial_invalid_tag_after_long_name.der

# pkcs7 signature file with invalid tag in signerInfo[2]
pkcs7_signerInfo_2_invalid_tag.der: pkcs7_data_3_signed.der
	cp $< $@
	echo 'a1' | xxd -p -r | dd of=$@ bs=1 seek=810 conv=notrunc
```
I think maybe `echo` in GNU Make doesn't have the `-e` option as the original didn't work for me, even though the same command works in a regular shell.

I got a certificate that was different from the original like so (generated certificate below):
![cert_diff](https://user-images.githubusercontent.com/70948878/221249111-414a186a-5e10-4362-b2b4-5003f88d6763.png)

Using `xxd` in reverse-mode instead solves the issue for me.

Reply:
Anyway, I can swap it out for a posixly correct solution.

Reply:
No, I'm using Ubuntu 20.04. The command `echo -en '\xa1'` works in a terminal, but when I run:
```sh
(cd tests/data_files && make pkcs7_signerInfo_1_serial_invalid_tag_after_long_name.der)
```
I get a certificate with a literal "-en \xa1" (no quotes, interestingly).

Reply:
Ah actually I think it's a bash feature, but Makefile uses `/bin/sh` by default.

Reply:
Ah, and ubuntu uses `/bin/dash` as `/bin/sh`. gotcha.

Reply:
I am sorry, I tried two ways locally: 
1. With the code from Bence, and add `SHELL := /bin/bash` to Makefile under tests/data_files.
2. With code changed by David, `echo 'a1' | xxd -p -r`.  
But seems I meet one stipid issue on my local, it seems the '3f' replaces 'a1', seems a complementary code exist there...

Reply:
For a similar problem in this Makefile with a previous PR for PKCS7, I used the Make built-in `printf` instead

Reply:
Hmm. `3f` is the ascii for `'?'`, which might imply that something in your setup is replacing any invalid unicode characters with `'?'`.

Reply:
> Hmm. `3f` is the ascii for `'?'`, which might imply that something in your setup is replacing any invalid unicode characters with `'?'`.


Seems you are right, and it always output `3f `whether I change it to` 'a1', 'a2' ,'a3', ...`  But I don't remember I did the setup manually.

Reply:
Not all implementations of `echo` support the `-e` option, and even for those who do, they don't all support the `\xNN` escape. It's a shell builtin, so it depends which shell make uses (normally `/bin/sh`) and not just which OS. On Ubuntu, `/bin/sh` is dash by default, and it supports `-e` but not `\xNN`.

You can use `printf '\241'` as a portable replacement. The downside is that you have to use octal: `\xNN` is also not portable with `printf`.

Alternatively, use `xxd`. It's not a basic Unix utility (it comes with Vim), but this makefile doesn't need to be widely portable, it only needs to run on developer machines. As a non-Vim-user, I'm perfectly fine with assuming the presence of `xxd` here. In any case `xxd` is already used.

Reply:
@xkqian A substitution of `?` tends to happen when printing to a terminal or converting to another encoding. The commands here should only manipulate binary data. What do you see when you run `echo 'a1' | xxd -p -r | od -t x1` ?

Reply:
> What do you see when you run `echo 'a1' | xxd -p -r | od -t x1` ?

I can see 
"
```
0000000 a1
0000001
```
"
exactly.  Seems it's my bad.  The bytes itself has no problem, but when I open the file with `VIM`, maybe some encoding convertion happen.  Once I open the file with "`vim -b`", then everything goes well.